src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -120,6 +120,20 @@ default Optional<T> getOptionalSingle(Event event) {
 	 */
 	boolean isSingle();
 
+	/**
+	 * Whether there's a possibility this could return a single value.
+	 * Designed for expressions that may return more than one value, but are equally appropriate to use where
+	 * only singular values are accepted, in which case a single value (out of all available values) will be returned.
+	 * An example would be functions that return results based on their inputs.
+	 * Ideally, this will return {@link #isSingle()} based on its known inputs at initialisation, but for some syntax
+	 * this may not be known (or a syntax may be intentionally vague in its permissible returns).
+	 * @return Whether this can be used by single changers
+	 * @see #getSingle(Event)
+	 */
+	default boolean canBeSingle() {
```

Comment:
Since https://github.com/SkriptLang/Skript/pull/6624 uses `mayReturn`, I think we should decide between `canX` vs `mayX` for consistency.

Reply:
Originally #6624 didn't use `may` but that was a requested change there. Would you like both to be `may` or both to be `can` ?

Reply:
I think we should stick with canX, i can't find any other mayX's but i did find multiple canX's.